hypervisor/qemu/qemu_amd64.go

```diff
@@ -56,6 +56,17 @@ func (qc *QemuContext) arguments(ctx *hypervisor.VmContext) []string {
 			"-kernel", boot.Kernel, "-initrd", boot.Initrd, "-append", cmdline)
 	}
 
+	if !hypervisor.Is9pfsSupported() {
+		params = append(params, "-device", "virtio-scsi-pci,id=scsi0,bus=pci.0,addr=0x6")
+		params = append(params, "-drive", "file=/tmp/"+ctx.Id+"/rootfs.img,format=qcow2,if=none,id=drive-scsi0-1-0-0")
+		params = append(params, "-device", "scsi-hd,bus=scsi0.0,channel=0,scsi-id=1,lun=0,drive=drive-scsi0-1-0-0,id=scsi0-1-0-0")
+		params = append(params, "-device", "virtio-serial-pci,id=virtio-serial0,bus=pci.0,addr=0x2", "-device", "virtio-scsi-pci,id=scsi1,bus=pci.0,addr=0x3")
```

Comment:
Disks should be inserted via special APIs()
the disk path should be specified in the container config.
This will result in doing disk hotplug.
(Avoiding disk hotplug is possible, but it requires some refactor to the code)

Reply:
By disk hotplug, are you suggesting that we could attach a disk to pod VM after it has started? If yes, could you help me point to that API? 

Also, the main motivation behind this PR is to make 9pfs optional. The way the code is structured right now makes all pod VMs to have 9pfs.

Reply:
yes. This is the way how hyperd/hyperctl/hyper.sh work on the images/volumes. We have the APIs ready, only except that runv-cli hasn't call these API so far, which it can be changed. hyperd/hyperctl/hyper.sh doesn't necessary depend on 9pfs, we are glad to make it allow to be configured out.

To hotplug add the rootfs image, you can just put the image info into RootVolume 
 of the ContainerDescription.
https://github.com/hyperhq/runv/blob/master/api/descriptions.pb.go#L99

The code for setting RootVolume might be here:
https://github.com/hyperhq/runv/blob/master/cli/container.go#L76-L77

Reply:
@harche  here when we [create container][runv_create], we created it with [a spec here by protobuf][container_spec], in which [defined a rootVolume][root_volume] with [VolumeDescription][VolumeDescription]:

```
message VolumeDescription {
    string name = 1;
    string source = 2;
    string format = 3; //"raw" (or "qcow2" later) for volume, "vfs" for dir path
    string fstype = 4; //"xfs", "ext4" etc. for block dev, or "dir" for dir path
    VolumeOption options = 8;
    bool dockerVolume = 9;
    bool readOnly = 10;
}
```

For block devices, you could specify `Source` as the path of device, Format as `raw`, and `Fstype` as `xfs` or `ext4` or any other fs in case the guest kernel supports it.

[runv_create]: https://github.com/hyperhq/runv/blob/6a1c8bf0790edce270db970195572227fc158a32/cli/container.go#L77
[container_spec]: https://github.com/hyperhq/runv/blob/master/api/descriptions.proto#L16
[root_volume]: https://github.com/hyperhq/runv/blob/master/api/descriptions.proto#L29
[VolumeDescription]: https://github.com/hyperhq/runv/blob/master/api/descriptions.proto#L46

Reply:
I am trying to implement the suggested approach above, but I am getting following error, https://pastebin.com/raw/UsfXNfdS

The change I made was, https://pastebin.com/raw/KLMxdJ63